src/extensionApi.ts

```diff
@@ -149,29 +149,35 @@ export class CommandHandler {
         if (context === undefined) {
             const serverId = await this.selectServer('Select server to show output channel');
             if (!serverId) return null;
-            context = this.serversData.serverStatus.get(serverId);
+            context = this.explorer.serverStatus.get(serverId);
         }
-        this.serversData.showOutput(context);
+        this.explorer.showOutput(context);
     }
 
-    public async restartServer(context?: Protocol.ServerState): Promise<void> {
+    public async restartServer(mode: string, context?: Protocol.ServerState): Promise<void> {
         if (context === undefined) {
             const serverId: string = await this.selectServer('Select server to restart', ServerState.STARTED);
             if (!serverId) return null;
-            context = this.serversData.serverStatus.get(serverId);
+            context = this.explorer.serverStatus.get(serverId);
         }
 
-        const params: Protocol.LaunchParameters = {
-            mode: 'run',
-            params: {
-                id: context.server.id,
-                serverType: context.server.type.id,
-                attributes: new Map<string, any>()
-            }
-        };
-
-        await this.client.getOutgoingSyncHandler().stopServerSync({ id: context.server.id, force: true });
-        await this.client.getOutgoingHandler().startServerAsync(params);
+        await this.client.getOutgoingSyncHandler().stopServerSync({ id: context.server.id, force: true })
+            .then(() => {
+                if (mode === 'debug') {
+                    return this.debugServer(context);
+                } else {
+                    const params: Protocol.LaunchParameters = {
+                        mode: mode,
```

Comment:
Wouldn't it be better if we type 'run' as mode or atleast check it before we create the params object?